### DISCUSSION TITLE
Allow extending `spacing` values with custom ones

### DISCUSSION THREAD

**Discussion by mcmxcdev** [mantinedev/mantine#3517] [Feature requests]
Based on our requirements from the design team, we need to have more spacing levels than the ones defined by Mantine.

As can be seen below, we would need a `xxs` and a `xxl` value to be part of our style system, but TypeScript disallows these. 

```
export const ZorroTheme: MantineThemeOverride = {
  spacing: {
    // @ts-expect-error Type '{ xxs: number; xs: number; sm: number; md: number; lg: number; xl: number; xxl: number; }' is not assignable to type 'DeepPartial<MantineSizes>'.
    xxs: 4,
    xs: 8,
    sm: 12,
    md: 16,
    lg: 20,
    xl: 24,
    xxl: 36,
  },
};

```

I haven't found a way to extend them properly and if I am not mistaken, it's only possible to extend the color palette currently.

**Comment by mattaningram** [comment]
I also need more spacing token options, finding Mantine's defaults in many aspects of theming to be very limited.

**Comment by xiaohanyu** [comment]
It seems that mantine v7 finally support extending `spacing` tokens for components:

Check:
- https://mantine.dev/core/container/#customize-sizes
- https://mantine.dev/styles/variants-sizes/#sizes-with-components-css-variables

**Comment by xiaohanyu** [comment]
New responsive breakpoints still not supported: https://discord.com/channels/854810300876062770/1215632487556972585/1216651247650607165
